web/src/lib/utils/asset-utils.ts

```diff
@@ -42,14 +43,27 @@ export const downloadBlob = (data: Blob, filename: string) => {
   URL.revokeObjectURL(url);
 };
 
+const downloadingArchives = new Set();
 export const downloadArchive = async (fileName: string, options: DownloadInfoDto) => {
+  const uniqueDownloadId = options.albumId ? `${options.albumId}` : `${options.assetIds?.sort().join(',')}`;
```

Comment:
@alextran1502, I need to test the case where we have a big list of assets to download how it will be the ${options.assetIds?.sort().join(',')}. Maybe I need to rethink it to avoid any perf issue.
Please don't merge this PR before I double check that. Thanks